openpype/plugins/publish/integrate_new.py

```diff
@@ -110,801 +156,494 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
+            # Single file transfer
+            src = os.path.join(stagingdir, fname)
+            transfers = [(src, dst)]
 
-        Args:
-            subset (dict): the registered subset of the asset
-            version_number (int): the version number
+        for key in self.db_representation_context_keys:
+            value = template_data.get(key)
+            if not value:
+                continue
+            repre_context[key] = template_data[key]
 
-        Returns:
-            dict: collection of data to create a version
-        """
+        # Explicitly store the full list even though template data might
+        # have a different value
+        if repre.get("udim"):
+            repre_context["udim"] = repre.get("udim")  # store list
 
-        return {"schema": "openpype:version-3.0",
-                "type": "version",
-                "parent": subset["_id"],
-                "name": version_number,
-                "data": data}
+        # Use previous representation's id if there is a name match
+        existing = existing_repres_by_name.get(repre["name"].lower())
+        if existing:
+            repre_id = existing["_id"]
+        else:
+            repre_id = ObjectId()
+
+        # Backwards compatibility:
+        # Store first transferred destination as published path data
+        # todo: can we remove this?
+        published_path = transfers[0][1]
+        repre["published_path"] = published_path  # Backwards compatibility
+
+        # todo: `repre` is not the actual `representation` entity
+        #       we should simplify/clarify difference between data above
+        #       and the actual representation entity for the database
+        data = repre.get("data", {})
+        data.update({'path': published_path, 'template': template})
+        representation = {
+            "_id": repre_id,
+            "schema": "openpype:representation-2.0",
+            "type": "representation",
+            "parent": version["_id"],
+            "name": repre['name'],
+            "data": data,
+            "dependencies": instance.data.get("dependencies", "").split(),
+
+            # Imprint shortcut to context for performance reasons.
+            "context": repre_context
+        }
+
+        # todo: simplify/streamline which additional data makes its way into
+        #       the representation context
+        if repre.get("outputName"):
+            representation["context"]["output"] = repre['outputName']
+
+        if is_sequence_representation and repre.get("frameStart") is not None:
+            representation['context']['frame'] = template_data["frame"]
+
+        return {
+            "representation": representation,
+            "anatomy_data": template_data,
+            "transfers": transfers,
+            # todo: avoid the need for 'published_files' used by Integrate Hero
+            # backwards compatibility
+            "published_files": [transfer[1] for transfer in transfers]
```

Comment:
Maybe if we refactor `files` on representation to be a dictionary as mentioned above, we could just add each file final destination in there, per file, that would also create a direct one to one mapping between source and dest files for any further plugins processing